samples/matter/common/src/bridge/ble_connectivity_manager.cpp

```diff
@@ -163,6 +163,11 @@ void BLEConnectivityManager::DiscoveryCompletedHandler(bt_gatt_dm *dm, void *con
 			provider->GetBLEBridgedDevice().mFirstConnectionCallback(
 				dm, discoveryResult, provider->GetBLEBridgedDevice().mFirstConnectionCallbackContext);
 			provider->ConfirmInitialConnection();
+		} else {
+			/* Provider has been already initialized and we need to parse the received data to attach the
+			 * gatt handle after Bridge reboot */
+			VerifyOrReturn(provider->ParseDiscoveredData(dm) == 0,
```

Comment:
I'm wondering if calling `ParseDiscoveredData` in two places (here and inside the `mFirstConnectionCallback`) is a good idea. Maybe we should remove calling it from the callback and call it here, unconditionally, before the `if (!provider->IsInitiallyConnected())` check.
It would make implementation cleaner and we would not push the responsibility for calling it to the user's application.

Reply:
Ok, I'm ok with it. Actually, I was wondering about that, and your comment assured me that do it in that way. :)

Reply:
Moreover, my comment there is redundant when we do it as you wrote because it seems natural that in DiscoevryCompleteHandeler we parse the discovered data.